File: ios/CameraManager.m
#import <React/RCTViewManager.h>
#import <React/RCTUIManager.h>

@interface RCT_EXTERN_MODULE(CameraManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(onReady, RCTBubblingEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onPhotoTaken, RCTBubblingEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onPhotoError, RCTBubblingEventBlock)
RCT_EXTERN_METHOD(takePhoto:(nonnull NSNumber *)reactTag)
@end

File: ios/CameraManager.swift
import Foundation

@objc(CameraManager)
class CameraManager: RCTViewManager {

    override func view() -> UIView! {
        return CameraView()
    }

    @objc func takePhoto(_ reactTag: NSNumber) {
        DispatchQueue.main.async {
            guard let view = self.bridge.uiManager.view(forReactTag: reactTag) as? CameraView else { return }
            view.takePhoto()
        }
    }

    @objc override static func requiresMainQueueSetup() -> Bool {
        return true
    }
}
